src/handling.rs

```diff
@@ -112,7 +113,7 @@ impl MessageHandler {
                                 error!("Unable to notify client {:?}", op)
                             });
                         let table_read = ktable.read().await;
-                        if payload.height > 0 {
+                        if payload.height > 0 && auto_broadcast {
```

Comment:
I'm thinking we could place the `if auto_broadcast` before the `ktable.read()` to prevent that block. Thoughts @herr-seppia?